openmc/weight_windows.py

```diff
@@ -410,3 +411,182 @@ def from_hdf5(cls, group, meshes):
+        for value in values:
+            yield value
+
+        # the remainder of the file can be read as
+        # sequential values
+        while True:
+            line = next(fh)
+            # skip empty or commented lines
+            if not line or line.startswith('c'):
+                continue
+            values = line.strip().split()
+            for value in values:
+                yield value
+
+
+def wwinp_to_wws(path):
+    """Creates WeightWindows classes from a wwinp file
+
+    Parameters
+    ----------
+    path : str or pathlib.Path object
+        Path to the wwinp file
+
+    Returns
+    -------
+    list of openmc.WeightWindows
+    """
+    # create generator for getting the next parameter from the file
+    wwinp = __wwinp_reader(path)
+
+    # first parameter, if, of wwinp file is unused
+    next(wwinp)
+
+    # check time parameter, iv
+    if int(float(next(wwinp))) > 1:
+        raise ValueError('Time-dependent weight windows '
+                         'are not yet supported.')
+
+    # number of particle types, ni
+    n_particle_types = int(float(next(wwinp)))
+
+    # read an indicator of the mesh type.
+    # this will be 10 if a rectilinear mesh
+    # and 16 for cylindrical or spherical meshes
+    mesh_chars = int(float(next(wwinp)))
+
+    if mesh_chars != 10:
+        # TODO: read the first entry by default and display a warning
+        raise NotImplementedError('Cylindrical and Spherical meshes '
+                                  'are not currently supported')
+
+    # read the number of energy groups for each particle, ne
+    n_egroups = [int(next(wwinp)) for _ in range(n_particle_types)]
+
+    # order that supported particle types will appear in the file
+    particle_types = ['neutron', 'photon']
+
+    # add particle to list if at least one energy group is present
+    particles = [p for e, p in zip(n_egroups, particle_types) if e > 0]
+    # truncate list of energy groups if needed
+    n_egroups = [e for e in n_egroups if e > 0]
+
+    if n_particle_types > 2:
+        msg = ('More than two particle types are present. '
+               'Only neutron and photon weight windows will be read.')
+        warnings.warn(msg)
+
+    # read total number of fine mesh elements in each coarse
+    # element (nfx, nfy, nfz)
+    n_fine_x = int(float(next(wwinp)))
+    n_fine_y = int(float(next(wwinp)))
+    n_fine_z = int(float(next(wwinp)))
+    header_mesh_dims = (n_fine_x, n_fine_y, n_fine_z)
+
+    # read the mesh origin: x0, y0, z0
+    llc = tuple(float(next(wwinp)) for _ in range(3))
+
+    # read the number of coarse mesh elements (ncx, ncy, ncz)
+    n_coarse_x = int(float(next(wwinp)))
+    n_coarse_y = int(float(next(wwinp)))
+    n_coarse_z = int(float(next(wwinp)))
+
+    # skip the value defining the geometry type, nwg, we already know this
+    # 1 - rectilinear mesh
+    # 2 - cylindrical mesh
+    # 3 - spherical mesh
+    mesh_type = int(float(next(wwinp)))
```

Comment:
It looks like in the cylindrical/spherical case, you have the number of coarse elements, then six floats describing vectors for the axes and **then** the geometry type. We should have an `if` here based on `mesh_chars`

Reply:
Good catch! Though in the case that we have extra values to read, we should have raised an exception before getting to this point.